PeerioTechnologies/peerio-desktop
# feat: DM placeholder when invite accepted

#### Relevant info and issue/PR links 
clubhouse: https://app.clubhouse.io/peerio/story/5590/desktop-placeholder-dm-for-new-contacts
icebear:  https://github.com/PeerioTechnologies/peerio-icebear/pull/189

#### Testing instructions  
See CH story for the Figma mockups. I found the flow kind of annoying to follow on Figma so here's a summary:
* `existinguser` invites `newuser` via email
* `newuser` signs up and confirms email address
* now both `existinguser` and `newuser` should have a "placeholder" DM in their ChatList
  * both have same design but with slightly different text
* this placeholder DM has two buttons:
  * dismiss -> placeholder DM removed from ChatList
  * message -> placeholder DM removed from ChatList and replaced with real DM
    * instead of the usual chat header "This is the beginning of your chat history...", newly created DM shows the same text as the placeholder DM, until app restart
* if pending DM exists and DM with that user is created through another means (e.g. "Direct messages +" button), this should dismiss the pending DM

**There was also a refactor of the "plus icon" component**, which affects several other views. These should look unchanged:
* ChatList: "Rooms +" and "Direct messages +" buttons
* Contacts view: "Contacts +" button
* ZeroChats: "+ Create a room" and "+ Create a direct message" buttons
* Onboarding: "Create a room" button
  * this one was using the .chat-item-add class but then hiding the icon, I'm not sure why. So I just removed the class.
----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
